packages/pharos/src/components/multiselect-dropdown/pharos-multiselect-dropdown.ts

```diff
@@ -0,0 +1,657 @@
+    );
+
+    const nextOption = options[nextOptionIndex];
+
+    nextOption.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    this._setHighlightedOption(nextOption.id);
+  }
+
+  private _selectHighlightedOption(): void {
+    const highlightedOption = this.renderRoot.querySelector(
+      '.multiselect-dropdown__option[highlighted]'
+    ) as HTMLLIElement;
+
+    if (highlightedOption) {
+      highlightedOption.click();
+    }
+  }
+
+  _handleFormdata(event: CustomEvent): void {
+    const { formData } = event;
+    if (!this.disabled) {
+      for (const option of this.selectedOptions) {
+        formData.append(this.name, option.value);
+      }
+    }
+  }
+
+  _handleFormReset(): void {
+    this.selectedOptions = this.options.filter((option) => option.selected);
+    this._pendingOptions = [];
+    this._searchValue = '';
+  }
+
+  private _getButtonDisplayText(): string {
+    if (this.selectedOptions.length === 0) {
+      return 'None Selected';
+    }
+    if (this.selectedOptions.length === this.options.length) {
+      return 'All Selected';
+    }
+    const selectedOptionsText = this.selectedOptions.map((option) => option.text).join(', ');
+    if (selectedOptionsText.length <= this.displayCharacterCount) {
+      return selectedOptionsText;
+    } else {
+      return `${this.selectedOptions.length} Selected`;
+    }
+  }
+
+  private _closeDropdown(): void {
+    this._searchValue = '';
+    this._open = false;
+    this._dropdown_button.focus();
+  }
+
+  private _toggleDropdown(): void {
+    if (this._open) {
+      this._closeDropdown();
+    } else {
+      this._open = true;
+    }
+    this._pendingOptions = this.selectedOptions;
+  }
+
+  private _handleDropdownButtonClick(): void {
+    if (this.disabled) {
+      return;
+    }
+    this._toggleDropdown();
+  }
+
+  private _handleDropdownKeydown(event: KeyboardEvent): void {
+    if (this.disabled) {
+      event.preventDefault();
+      return;
+    }
+    if (event.key === 'Enter' || event.key === 'Space') {
+      this._handleDropdownButtonClick();
+    }
+  }
+
+  private _handleApplyClick(): void {
+    this.selectedOptions = [...this._pendingOptions];
+    this._closeDropdown();
+    this.onChange();
+  }
+
+  private _handleCancelClick(): void {
+    this._pendingOptions = [];
+    this._closeDropdown();
+    this.onChange();
+  }
+
+  /**
+   * Normalizes a string in the following order:
+   *
+   * 1 - Applies https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
+   *     with the 'NFKD' parameters for compatibility
+   * 2 - Removes all diacritic characters using a global regex
+   * 3 - Lower case all characters
+   *
+   */
+  private _normalizeString(textString: string) {
+    return textString
+      .normalize('NFKD')
+      .replace(/[\u0300-\u036f]/g, '')
+      .toLowerCase();
+  }
+
+  // Renders the checkbox icon used in the dropdown options from an svg
+  private _renderCheckboxIcon(
```

Comment:
Is there a reason these couldn't be `pharos-checkbox`es? If so, that feels like it's worth a code comment here.

Reply:
https://github.com/ithaka/pharos/pull/962/commits/28bf3f26619d045940775b504ed860761ff47147